components/ias-adapter/internal/service/tenant_mapping.go

```diff
@@ -0,0 +1,126 @@
+package service
+
+import (
+	"context"
+
+	"github.com/kyma-incubator/compass/components/ias-adapter/internal/errors"
+	"github.com/kyma-incubator/compass/components/ias-adapter/internal/logger"
+	"github.com/kyma-incubator/compass/components/ias-adapter/internal/service/ias"
+	"github.com/kyma-incubator/compass/components/ias-adapter/internal/storage/postgres"
+	"github.com/kyma-incubator/compass/components/ias-adapter/internal/types"
+)
+
+type TenantMappingsStorage interface {
+	UpsertTenantMapping(ctx context.Context, tenantMapping types.TenantMapping) error
+	ListTenantMappings(ctx context.Context, formationID string) (map[string]types.TenantMapping, error)
+	DeleteTenantMapping(ctx context.Context, formationID, applicationID string) error
+}
+
+type IASService interface {
+	GetApplication(ctx context.Context, iasHost, clientID string) (types.Application, error)
+	UpdateApplicationConsumedAPIs(ctx context.Context, data ias.UpdateData) error
+}
+
+type TenantMappingsService struct {
+	Storage    TenantMappingsStorage
+	IASService IASService
+}
+
+func (s TenantMappingsService) ProcessTenantMapping(ctx context.Context, tenantMapping1 types.TenantMapping) error {
+	log := logger.FromContext(ctx)
+
+	formationID := tenantMapping1.FormationID
+	tenantMappings, err := s.Storage.ListTenantMappings(ctx, formationID)
+	if err != nil {
+		log.Err(err).Send()
+		return errors.Newf("failed to get tenant mappings for formation '%s': %w", formationID, err)
+	}
```

Comment:
Usually an error is either logged or returned. In this case the returned error has more context (what operation has failed + the reason it failed) so logging it (possibly in the layer above (where it is called)) seems beneficial.

Same below (lines 51, and 61).